Added toggle to enable H2 heading underlines

Tejeev opened the pull request:
I love the theme but I find my notes much easier to read when the H2 headings have underlines.  

If you're cool with it, I'd like to add the toggleable option to have underlines under H2 headings with this commit.   

I've left the option off by default so as to not affect users who do not wish for the change/option.

Tejeev committed:
Toggle to enable H2 heading underlines

I love the theme but I find my notes much easier to read when the H2 headings have underlines.  If you're cool with it, I'd like to add the toggleable option to have underlines under H2 headings with this commit.  
I've left the option off by default so as to not affect users who do not wish for the change/option.

Tejeev commented:
Worth noting that I'm not very knowledgeable with CSS or skilled in styling, so I borrowed the underline from the Things theme.  Do I need to make any changes / rewrite of the little block/how it's done, or is it acceptable?

FireIsGood commented:
The styles seem alright, though there seems to be an unused `width: 100%`. I'm going to move over the styles to the spots that make more sense for end users (header setting will go into its own heading under Elements with the other element modifications and the styles under the style setting related toggles section below the comment). I'll have to clean up some other ordering but I'll do that later on the main branch.

I'm considering adding other header level toggles for lines, do you think it would be better to have all headers able to get an underline like `h2`? If so, this might be a setting to be nested in another level of Style setting header (e.g. `Elements > Headers > Underlines`) in the case of future changes.

In any case, thanks for the pull request! I'm making some more minor adjustments (setting naming scheme, etc.) and will merge soon.

FireIsGood committed:
Change header 2 setting location

The h2 style has been moved down to inside the style settings toggle
area to be easier to find and the setting has been renamed and moved to
a header category under the Elements header.

Tejeev commented:
@FireIsGood Sweet, I love your edits. I thought where the setting ended up was a bad spot but hadn't really figured out how to do things better.  

I do agree that it may be useful to have the option to underline any of the headers.  Doesn't make sense to have the option on just one level since everyone has their own needs.  

Are you adding more formatting options?  I was actually just looking to change the color of **bold** because I've decided to use a bold H1 as my page titles and regular H1 as the first level of headings, and realized it along with heading styling wasn't exposed in the Style Settings.  Was considering just modifying the theme to expose all of them so I don't have to modify CSS every time, but then I'm basically forking and I don't feel confident I'd do it in the best way :P

FireIsGood commented:
Yeah, I'll try and add options for all the headers when I get some time.

For formatting with headers and bold, I could add a toggle for the emphasis stuff to disable the color changes in headings and/or all text, though I think semantically it should always be a single `h1` and all other cases should be `h2` or below. If you want, I could write a snippet really quick to fix header bold colors for you use case, though!

If you want to ask more questions more synchronously, I'm active on the Obsidian Members Group Discord at `@fireisgood`, though here is fine as well.

Tejeev commented:
I'll hop on discord.  Just to round out the convo on this thread, I was just thinking of making bold a green color I'd choose after some experimentation.
I was leaning towards re-using the H1 because it feels kind of wasteful to use it only once in a note, but that's how we used it back in the html days, and it helps keep things organized, so that's a fair pushback.  Really I was being lazy.  I just wanted to shift the current heading styles down one and slip in a new one for the actual page titles, and figured it was easier to just bold the H1 at the top of the page... but I wanted to chose my own color themed by vault for that.
